### PR TITLE
Implement #2750: Run `black`+`isort` on `gen_exports` results

### PR DESCRIPTION
**Pull request by TeamSpen210**
This makes `gen_exports` run both linters on the generated files, improving the formatting a little. 

I ran `black` via subprocess because its API doesn't provide a nice way to fetch configuration from `pyproject.toml` etc to maintain consistency, while also processing an in-memory string. Still doesn't remove the redundant parentheses, but oh well.

### CODE REVIEW

**Review by jakkdl — APPROVED** [review]
Looks great!

**Comment by TeamSpen210** [comment]
Adding this caused all the pypy tests to fail, since we don't even install black/isort there. I just skipped the tests there, the module is an internal tool that only needs to run on one version.

**Comment by jakkdl** [comment]
needs pragma: skip on the exceptions

**Comment by TeamSpen210** [comment]
Hmm actually I could just add tests for those cases - mock out `isort` to produce an error, send in an invalid file to make `black` fail.

**Comment by jakkdl** [comment]
> Hmm actually I could just add tests for those cases - mock out `isort` to produce an error, send in an invalid file to make `black` fail.

sure .. but do we need thorough tests on an internal code generation tool? :p

File: trio/_tools/gen_exports.py [hunk 1]
```diff
@@ -19,22 +21,21 @@
...
-# fmt: off
 from __future__ import annotations
 
 from ._ki import LOCALS_KEY_KI_PROTECTION_ENABLED
 from ._run import GLOBAL_RUN_CONTEXT
 """
 
-FOOTER = """# fmt: on
+FOOTER = """
 """
```
Comment (A5rocks) [inline trio/_tools/gen_exports.py:39]: Why keep the newline?

File: trio/_tools/gen_exports.py [hunk 2]
```diff
@@ -104,6 +105,40 @@ def create_passthrough_args(funcdef: ast.FunctionDef | ast.AsyncFunctionDef) ->
...
 
 
+def run_linters(file: File, source: str) -> str:
+    """Run isort and black on the specified file, returning the new source.
+
+    :raises ValueError: If either failed.
+    """
+    # Black has an undocumented API, but it doesn't easily allow reading configuration from
+    # pyproject.toml, and simultaneously pass in / receive the code as a string.
+    # https://github.com/psf/black/issues/779
```
Comment (A5rocks) [inline trio/_tools/gen_exports.py:115]: FYI I played around with black's API (just using `format_file_contents` with `Mode` which was just a retyped config file; just line length and preview though) but a dev on black told me to use the CLI interface like you're doing instead :^)

File: trio/_tools/gen_exports.py [hunk 3]
```diff
@@ -104,6 +105,40 @@ def create_passthrough_args(funcdef: ast.FunctionDef | ast.AsyncFunctionDef) ->
...
+    # isort does have a public API, makes things easy.
+    try:
+        isort_res = isort.api.sort_code_string(
+            result.stdout,
+            file_path=file.path,
+        )
+    except isort.exceptions.ISortError as exc:
+        print("Failed to run isort!")
+        traceback.print_exception(type(exc), exc, exc.__traceback__)
+        sys.exit(1)
```
Comment (A5rocks) [inline trio/_tools/gen_exports.py:138]: Why can't we re-raise? Or even `raise RuntimeError("Failed to run isort!") from exc`!

**Review by A5rocks — COMMENTED** [review]
Couple of nitpicks, looks good otherwise.

File: trio/_tools/gen_exports.py (hunk 3 above)
Comment (TeamSpen210) [reply trio/_tools/gen_exports.py:138]: I don't think that results in a nonzero exit code, I wanted to make sure CI etc fails.

File: trio/_tools/gen_exports.py (hunk 1 above)
Comment (TeamSpen210) [reply trio/_tools/gen_exports.py:39]: Good question.

File: trio/_tools/gen_exports.py (hunk 3 above)
Comment (A5rocks) [reply trio/_tools/gen_exports.py:138]: FWIW it does, it seems!

```py
a5@hostname:/tmp$ python3 t.py
Traceback (most recent call last):
  File "t.py", line 1, in <module>
    raise Exception("hi!")
Exception: hi!
a5@hostname:/tmp$ echo $?
1
```
(at least on my Ubuntu WSL env...)

**Comment by A5rocks** [comment]
builds.sr.ht is back!?!

**Comment by jakkdl** [comment]
I'll merge this unless somebody thinks otherwise, re-raising the exception instead of printing traceback seems like a minor enough problem not worth stalling over

**Pull request merged**